### ISSUE TITLE
Does Ontop support OWL inference?

### ISSUE THREAD

**Issue by AresEkb** [ontop/ontop#300]
Hi

I developed a trivial ontology:

```
Prefix(:=<http://www.semanticweb.org/dnikiforov/ontologies/2019/7/untitled-ontology-4#>)
Prefix(owl:=<http://www.w3.org/2002/07/owl#>)
Prefix(rdf:=<http://www.w3.org/1999/02/22-rdf-syntax-ns#>)
Prefix(xml:=<http://www.w3.org/XML/1998/namespace>)
Prefix(xsd:=<http://www.w3.org/2001/XMLSchema#>)
Prefix(obda:=<https://w3id.org/obda/vocabulary#>)
Prefix(rdfs:=<http://www.w3.org/2000/01/rdf-schema#>)


Ontology(<http://www.semanticweb.org/dnikiforov/ontologies/2019/7/untitled-ontology-4>

Declaration(Class(:City))
Declaration(Class(:CityKind))
Declaration(Class(:Village))
Declaration(Class(:VillageKind))
Declaration(ObjectProperty(:hasKind))
Declaration(ObjectProperty(:kindOf))
Declaration(DataProperty(:name))
Declaration(DataProperty(:okato))
Declaration(DataProperty(:oktmo))
Declaration(DataProperty(:shortName))
############################
#   Object Properties
############################

# Object Property: :hasKind (:hasKind)

InverseObjectProperties(:hasKind :kindOf)
ObjectPropertyDomain(:hasKind :City)
ObjectPropertyRange(:hasKind :CityKind)

# Object Property: :kindOf (:kindOf)

SubObjectPropertyOf(:kindOf owl:topObjectProperty)


############################
#   Data Properties
############################

# Data Property: :name (:name)

SubDataPropertyOf(:name owl:topDataProperty)
DataPropertyDomain(:name :City)
DataPropertyRange(:name xsd:string)

# Data Property: :okato (:okato)

DataPropertyDomain(:okato :City)
DataPropertyRange(:okato xsd:string)

# Data Property: :oktmo (:oktmo)

DataPropertyDomain(:oktmo :City)
DataPropertyRange(:oktmo xsd:string)

# Data Property: :shortName (:shortName)

SubDataPropertyOf(:shortName owl:topDataProperty)
DataPropertyDomain(:shortName :CityKind)
DataPropertyRange(:shortName xsd:string)



############################
#   Classes
############################

# Class: :Village (:Village)

SubClassOf(:Village ObjectSomeValuesFrom(:hasKind :VillageKind))

# Class: :VillageKind (:VillageKind)

SubClassOf(:VillageKind :CityKind)
)
```

with the following mappings:

```
[PrefixDeclaration]
:		http://www.semanticweb.org/dnikiforov/ontologies/2019/7/untitled-ontology-4#
owl:		http://www.w3.org/2002/07/owl#
rdf:		http://www.w3.org/1999/02/22-rdf-syntax-ns#
xml:		http://www.w3.org/XML/1998/namespace
xsd:		http://www.w3.org/2001/XMLSchema#
obda:		https://w3id.org/obda/vocabulary#
rdfs:		http://www.w3.org/2000/01/rdf-schema#

[MappingDeclaration] @collection [[
mappingId	CityMapping
target		:City/{Id} a :City ; :name {Name}^^xsd:string ; :okato {Code}^^xsd:string ; :oktmo {AdministrativeCode}^^xsd:string ; :hasKind :CityKind/{KindId} . 
source		SELECT [Id]
			,[Name]
			,[Code]
			,[AdministrativeCode]
			,[KindId]
			FROM [City]

mappingId	CityKindMapping
target		:CityKind/{Id} a :City ; :shortName {ShortName}^^xsd:string . 
source		SELECT [Id], [ShortName]
			FROM [CityKind]

mappingId	VillageKindMapping
target		:VillageKind/{Id} a :VillageKind ; :shortName {ShortName}^^xsd:string . 
source		SELECT [Id], [ShortName]
			FROM [CityKind]
			WHERE [Id] = 317
]]
```

Trivial SPARQL queries works fine. However queries involving OWL inference returns nothing:
```
PREFIX : <http://www.semanticweb.org/dnikiforov/ontologies/2019/7/untitled-ontology-4#>
SELECT * WHERE { ?k :kindOf ?c . }
```
```
PREFIX : <http://www.semanticweb.org/dnikiforov/ontologies/2019/7/untitled-ontology-4#>
SELECT * WHERE { ?c a :Village . }
```

I test the queries on Ontop Server. If I execute this non-trivial queries in Protege, then an SQL Translation window is empty. It seems that it doesn't generate any SQL query. There is no any errors in a log.

I see the following message in the log:
```
Pre-computing inferences:
    - class hierarchy
    - object property hierarchy
    - data property hierarchy
    - class assertions
    - object property assertions
    - same individuals
```

Does Ontop support InverseObjectProperty axiom? As I can see in [this example](https://github.com/ontop/ontop/wiki/Example_MovieOntology) `isActorIn` property is inverse of `hasActor` property, and `hasActor` is a super-property of 2 mapped properties `hasMaleActor` and `hasActress`. So I guess that inverse properties should be infered. But why it's not the case in my example?

**Comment by ghxiao** [comment]
Hi, the issue related to inverse properties should be fixed in the release of 3.0.1. Please give it a try.

For the query about instances of "Village", I don't see how they could get any answer. The axiom "SubClassOf(:Village ObjectSomeValuesFrom(:hasKind :VillageKind))" and others could not generate an instance of Village.

**Comment by ghxiao** [comment]
Hi, I close the issue for now, but feel free to reopen it if you still have any problem.
